### ISSUE TITLE
ICH AC97 Support

### ISSUE THREAD

**Issue by shockdesign** [crazii/SBEMU#5]
Device: HP Compaq nx8220
OS: FreeDOS

```
Found sound card: ICH AC97
Real mode support: enabled.
Protected mode support: enabled.
OPL3 emulation enabled at port 388h.
Sound Blaster 16 emulation enabled at Adress: 220, IRQ: 7, DMA: 1
```

However I cannot get anything to play sound.
Impulse tracker detects sound card but doesn't play back any music.
I've tried Adlib Tracker II and RAD Tracker as well to test out adlib and nothing either.

What steps can I take to start debugging any of this?

**Comment by MFDGaming** [comment]
> Device: HP Compaq nx8220 OS: FreeDOS
> 
> ```
> Found sound card: ICH AC97
> Real mode support: enabled.
> Protected mode support: enabled.
> OPL3 emulation enabled at port 388h.
> Sound Blaster 16 emulation enabled at Adress: 220, IRQ: 7, DMA: 1
> ```
> 
> However I cannot get anything to play sound. Impulse tracker detects sound card but doesn't play back any music. I've tried Adlib Tracker II and RAD Tracker as well to test out adlib and nothing either.
> 
> What steps can I take to start debugging any of this?

try reserving IRQ7 in the bios or try running sbemu with IRQ5

**Comment by megatog615** [comment]
Doesn't work here on my ICH AC97 chip. I have double checked that the IRQ/DMA is free and disabled unnecessary devices in the BIOS to free up resources. HWINFO and CheckIt both show that the IRQ and DMA are free before I activate SBEMU.

Duke3D Setup complains about an invalid or conflicting DMA channel, and music is completely silent. This is on a Panasonic Toughbook CF-M34 Pentium III system with AC'97 audio.

**Comment by MFDGaming** [comment]
> Doesn't work here on my ICH AC97 chip. I have double checked that the IRQ/DMA is free and disabled unnecessary devices in the BIOS to free up resources. HWINFO and CheckIt both show that the IRQ and DMA are free before I activate SBEMU.
> 
> Duke3D Setup complains about an invalid or conflicting DMA channel, and music is completely silent. This is on a Panasonic Toughbook CF-M34 Pentium III system with AC'97 audio.

does it say the driver loaded or does it have error messages?

**Comment by MFDGaming** [comment]
Ive personally only tested it on a HDA chip with a core i5 10400F and a core i7 2nd gen,
i may test it on a board with an AC97 and a pentium4 or a celeron

**Comment by megatog615** [comment]
> > Doesn't work here on my ICH AC97 chip. I have double checked that the IRQ/DMA is free and disabled unnecessary devices in the BIOS to free up resources. HWINFO and CheckIt both show that the IRQ and DMA are free before I activate SBEMU.
> > Duke3D Setup complains about an invalid or conflicting DMA channel, and music is completely silent. This is on a Panasonic Toughbook CF-M34 Pentium III system with AC'97 audio.
> 
> does it say the driver loaded or does it have error messages?

No, no errors. It does say "Driver Loaded." It gives IRQ, DMA, etc information. I can change these on the fly by running the program again as expected.

**Comment by MFDGaming** [comment]
> > > Doesn't work here on my ICH AC97 chip. I have double checked that the IRQ/DMA is free and disabled unnecessary devices in the BIOS to free up resources. HWINFO and CheckIt both show that the IRQ and DMA are free before I activate SBEMU.
> > > Duke3D Setup complains about an invalid or conflicting DMA channel, and music is completely silent. This is on a Panasonic Toughbook CF-M34 Pentium III system with AC'97 audio.
> > 
> > 
> > does it say the driver loaded or does it have error messages?
> 
> No, no errors. It does say "Driver Loaded." It gives IRQ, DMA, etc information. I can change these on the fly by running the program again as expected.

In duke I remember I had to change some setting in order to get it working, doom worked out right away though

**Comment by androx74** [comment]
try to use a sbpro string on autoexec.bat and test again sb16 has a lot of trouble working

**Comment by megatog615** [comment]
> try to use a sbpro string on autoexec.bat and test again sb16 has a lot of trouble working

No difference.

**Comment by shockdesign** [comment]
As an update, beta 3 seems to play Adlib fine on my hardware now when beta 2 didn't, haven't done a thorough test on SB support yet.
